docs/require-by-string-aliases.md

```diff
@@ -0,0 +1,265 @@
+
+```
+luau-paths-project
+├── .luaurc
+├── dependencies
+│   └── dependency.luau
+└── src
+    └── module.luau
+```
+
+If `.luaurc` contained the following `paths` array:
+```json
+{
+    "paths": ["./dependencies"]
+}
+```
+
+Then, `module.luau` could simply require `dependency.luau` like this:
+```lua
+local dependency = require("dependency")
+
+-- Instead of: require("../dependencies/dependency")
+```
+
+Using the `paths` array allows Luau developers to organize their projects however they like without compromising code readability.
+
+##### Current limitations of aliases
+
+- Aliases cannot reference other aliases. (However, this is compatible with this proposal and will likely be implemented in the future.)
+- Alias names cannot contain the directory separators `/` and `\`.
+- Aliases can only occur at the beginning of a path.
+
+##### Configuring alias maps: .luaurc
+
+As part of this proposal, alias maps will be configured in `.luaurc`, which follows a JSON-like syntax.
+
+Proposed structure of an alias map:
+```json
+{
+    "aliases": {
+        "alias1": "/path/of/alias1",
+        "alias2": "/path/of/alias2"
+    }
+}
+```
+
+Missing aliases in `.luaurc` are inherited from the alias maps of any parent directories, and fields can be overridden.
+
+Additionally, if an alias is bound to a relative path, the path will be evaluated relative to the `.luaurc` file in which the alias was defined.
+
+Finally, providing support for alias maps within the Roblox engine is out of the scope of this RFC but is being considered internally.
+
+### Symlinks
+
+Symlinks carry some security concerns; for example, a link's target might exist outside of the project folder in which the link was defined. For the first version of this implementation, symlinks will not be supported and will be treated as ordinary files.
+
+If we do implement symlinks in the future, we will likely use our own limit to the number of symlinks to "follow through" for cross-platform compatibility. It is also possible that we will add a new configurable property in `.luaurc` that will allow developers to toggle whether or not to resolve symlinks.
+
+Similar examples:
+- https://www.typescriptlang.org/tsconfig#preserveSymlinks
+- https://webpack.js.org/configuration/resolve/#resolvesymlinks
+
+## Use Cases
+### Alias map
+
+Using alias maps, Luau developers can require globally installed Luau libraries in their code without needing to specify their locations in Luau scripts.
+
+```
+luau-aliases-project
+├── .luaurc
+└── src
+    └── module.luau
+```
+
+For example, if we wanted to require `Roact` in `module.luau`, we could add the following alias to `.luaurc`:
+
+```json
+{
+    "aliases": {
+        "Roact": "/Users/johndoe/LuauLibraries/Roact/src"
+    }
+}
+```
+
+Then, we could simply write the following in `module.luau`, and everything would work as intended:
+```lua
+local Roact = require("@Roact")
```

Comment:
Roact in the alias map is pointing to a directory correct? What does the require of a directory point to - is it an `init.luau` file inside that directory? 

(Maybe this is actually a question for #6)